src/test/regress/expected/multi_colocation_utils.out

```diff
@@ -763,20 +921,93 @@ SELECT * FROM pg_dist_colocation
+ table2_groupf     |            6
+ table1_group_none |            7
+ table2_group_none |            8
+(13 rows)
+
+-- move the all tables in colocation group 5 to colocation group 7
+SELECT mark_tables_colocated('table1_group_none', ARRAY['table1_groupE', 'table2_groupE', 'table3_groupE']);
+ mark_tables_colocated 
+-----------------------
+ 
+(1 row)
+
+-- move a table with a colocation id which is already not in pg_dist_colocation
+SELECT mark_tables_colocated('table1_group_none', ARRAY['table2_group_none']);
+ mark_tables_colocated 
+-----------------------
+ 
+(1 row)
+
+-- check metadata to see that unused colocation group is deleted
+SELECT * FROM pg_dist_colocation
+    WHERE colocationid >= 1 AND colocationid < 1000
+    ORDER BY colocationid;
+ colocationid | shardcount | replicationfactor | distributioncolumntype 
+--------------+------------+-------------------+------------------------
+            2 |          2 |                 1 |                     23
+            3 |          2 |                 2 |                     25
+            4 |          4 |                 2 |                     23
+            6 |          1 |                 2 |                     23
+(4 rows)
+
+SELECT logicalrelid, colocationid FROM pg_dist_partition
+    WHERE colocationid >= 1 AND colocationid < 1000
+    ORDER BY colocationid, logicalrelid;
+   logicalrelid    | colocationid 
+-------------------+--------------
+ table1_groupb     |            2
+ table2_groupb     |            2
+ table1_groupc     |            3
+ table2_groupc     |            3
+ table1_groupd     |            4
+ table2_groupd     |            4
+ table1_groupf     |            6
+ table2_groupf     |            6
+ table1_groupe     |            7
+ table2_groupe     |            7
+ table3_groupe     |            7
+ table1_group_none |            7
+ table2_group_none |            7
+(13 rows)
+
+-- try ro colocate different replication models
+CREATE TABLE table1_groupG ( id int );
+SELECT create_distributed_table('table1_groupG', 'id');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
+
+-- update replication model
+UPDATE pg_dist_partition SET repmodel = 's' WHERE logicalrelid = 'table1_groupG'::regclass;
+CREATE TABLE table2_groupG ( id int );
+SELECT create_distributed_table('table2_groupG', 'id', colocate_with => 'table1_groupG');
+ERROR:  cannot colocate tables table1_groupg and table2_groupg
+DETAIL:  Replication models don't match for table1_groupg and table2_groupg.
+CREATE TABLE table2_groupG ( id int );
+ERROR:  relation "table2_groupg" already exists
+SELECT create_distributed_table('table2_groupG', 'id', colocate_with => 'NONE');
+ create_distributed_table 
+--------------------------
+ 
+(1 row)
 
+SELECT mark_tables_colocated('table1_groupG', ARRAY['table2_groupG']);
+ERROR:  cannot colocate tables table1_groupg and table2_groupg
+DETAIL:  Replication models don't match for table1_groupg and table2_groupg.
```

Comment:
Can we drop unnecessary tables at the end of the test if they are not used by other tests? Otherwise test space becomes too crowded.